central/delegatedregistryconfig/convert/convert.go

```diff
@@ -68,3 +70,63 @@ func APIToStorage(from *v1.DelegatedRegistryConfig) *storage.DelegatedRegistryCo
 		Registries:       regs,
 	}
 }
+
+// APIToInternalAPI converts a delegated registry config from the type used by the GRPC/REST API
+// to the type used by central/sensor inner apis
+func APIToInternalAPI(from *v1.DelegatedRegistryConfig) *central.DelegatedRegistryConfig {
+	if from == nil {
+		return nil
+	}
+
+	var regs []*central.DelegatedRegistryConfig_DelegatedRegistry
+
+	if len(from.Registries) > 0 {
+		regs = make([]*central.DelegatedRegistryConfig_DelegatedRegistry, len(from.Registries))
+
+		for i, reg := range from.Registries {
+			regs[i] = &central.DelegatedRegistryConfig_DelegatedRegistry{
+				ClusterId:    reg.ClusterId,
+				RegistryPath: reg.RegistryPath,
+			}
+		}
+	}
```

Comment:
The change yields `index out of range [0] with length 0`.

Moving `len(registries)` to `size` instead of `cap` works, however `regs` will now always be non-nil even if `len(registries) == 0`. My preference would be to keep this as is so that not creating a zero length slice when not needed - if you feel strongly about it will make the change. 

Let me know!